txstratum/api.py

```diff
@@ -112,21 +112,24 @@ async def submit_job(self, request: web.Request) -> web.Response:
             self.log.debug('non-standard-tx', data=data)
             return web.json_response({'error': 'non-standard-tx'}, status=400)
 
-        # Check for banned inputs.
-        for txin in tx.inputs:
-            if txin.tx_id in self.banned_tx_ids:
-                self.log.info('txin-banned', data=data)
+        if self.toiclient:
+            # Check for banned txs and addresses.
+            txs: Set[str] = set()
+            addrs: Set[str] = set()
+            for txin in tx.inputs:
+                txs.add(txin.tx_id.decode())
```

Comment:
`txin.tx_id.decode()` is working fine? At first I thought we should do `txin.tx_id.hex()`

Reply:
It shouldn't work a lot of the times (not every byte sequence is a valid str, especially when using decode without arguments, which will default to utf-8 which has more precise rules).

I'd prefer to either use a `Set[bytes]`, or @pedroferreira1's suggestion of using `txin.tx_id.hex()`.

Reply:
Checked and the correct one should `.hex()`